docs/4.0-migration-guide/pk.md

```diff
@@ -0,0 +1,85 @@
+
+### Changes to PK metadata
+
+The type `mbedtls_pk_type_t` has been removed from the API. This type could convey different kinds of information, depending on where it was used:
+
+* Information about the representation of a PK context (distinguishing between transparent and opaque objects containing the same type of key). This information is no longer exposed.
+
+* Which signature algorithm to use (distinguishing between PKCS#1v1.5 and PSS for RSA keys), mostly for the sake of X.509. This information is now represented as `mbedtls_pk_sigalg_t`, using enum constants with tweaked names:
+
+    | Old `mbedtls_pk_type_t` name | New `mbedtls_pk_sigalg_t` name |
+    | ---------------------------- | ------------------------------ |
+    | `MBEDTLS_PK_NONE` | `MBEDTLS_PK_SIGALG_NONE` |
+    | `MBEDTLS_PK_RSA` | `MBEDTLS_PK_SIGALG_RSA_PKCS1V15` |
+    | `MBEDTLS_PK_RSASSA_PSS` | `MBEDTLS_PK_SIGALG_RSA_PSS` |
+    | `MBEDTLS_PK_ECDSA` or `MBEDTLS_PK_ECKEY` | `MBEDTLS_PK_SIGALG_ECDSA` |
+
+* Policy information after parsing a key (only relevant for ECC keys marked as ECDH-only). This is now tracked internally and reflected in rejecting the key for signature or verification in `mbedtls_pk_get_psa_attributes()` and in operation functions.
+  <!-- Untested, see https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/206 -->
+
+As a consequence, the functions ``mbedtls_pk_get_type()`, `mbedtls_pk_get_name()` and `mbedtls_pk_info_from_type()` and have been removed. The type `mbedtls_pk_info_t` is no longer part of the API.
+
+The function `mbedtls_pk_get_len()` has also been removed. It was not very meaningful since it did not convey the length of the key representation, but the size in bytes of the representation of one number associated with the key. As before, you can use `mbedtls_pk_get_bitlen()` to get the key size in the usual cryptographic sense. The size of a formatted key representation depends on the format, and there is no API function to determine it. (For the short export formats of PSA, you can use macros such as `PSA_EXPORT_KEY_OUTPUT_SIZE`, `PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE`, `PSA_EXPORT_KEY_PAIR_MAX_SIZE`, `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE`.)
+
+### Checking the capabilities of a PK context
+
+The function `mbedtls_pk_can_do()`, based on the polysemic `mbedtls_pk_type_t`, has been removed. The function `mbedtls_pk_can_do_ext()`, based on PSA metadata, has been renamed to `mbedtls_pk_can_do_psa()`. It now supports public-key operations as well as private-key operations.
+
+### Loss of policy information for RSA keys
+
+The PK module no longer partially keeps track of whether an RSA key is inteded for use with PKCS#1v1.5 or PKCS#1v2.1 (PSS or OAEP) algorithms. Functions that consume a PK object (`mbedtls_pk_get_psa_attributes()`, `mbedtls_pk_sign()`, `mbedtls_pk_verify()`) now always default to PKCS#1v1.5, except for wrapped PSA keys where they use the primary algorithm in the key's policy.
```

Comment:
Minor: perhaps s/wrapped PSA keys/contexts populated with `mbedtls_pk_wrap_psa()`/ to make it extra clear that it's not about whether the key is internally stored as a PSA key or not. (I know the document already said earlier that's unspecified, so a logical consequence is that it can't influence observable behaviour, but still.)